### PR TITLE
feat(world): worldProxy emits hello events when upgraded

### PR DESCRIPTION
**Pull request by yonadaaa**
continued from https://github.com/latticexyz/mud/pull/2632
This means we can detect when the MUD protocol version has changed

### CODE REVIEW

File: packages/world/src/WorldProxy.sol [hunk 1]
```diff
@@ -44,6 +41,8 @@ contract WorldProxy is Proxy {
     StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
 
     emit IERC1967.Upgraded(newImplementation);
+    emit IStoreEvents.HelloStore(STORE_VERSION);
+    emit IWorldEvents.HelloWorld(WORLD_VERSION);
```
Comment (holic) [inline packages/world/src/WorldProxy.sol:45]: These need to come from the implementation, not the hardcoded constant

File: packages/world/src/WorldProxy.sol [hunk 2]
```diff
@@ -44,6 +40,10 @@ contract WorldProxy is Proxy {
     StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
 
     emit IERC1967.Upgraded(newImplementation);
+
+    IBaseWorld world = IBaseWorld(newImplementation);
+    emit IStoreEvents.HelloStore(world.storeVersion());
+    emit IWorldEvents.HelloWorld(world.worldVersion());
```
Comment (holic) [inline packages/world/src/WorldProxy.sol:46]: although this isn't a place we need to really gas optimize, I kinda like what you had before 
```suggestion
    emit IStoreEvents.HelloStore(IBaseWorld(world).storeVersion());
    emit IWorldEvents.HelloWorld(IBaseWorld(world).worldVersion());
```

Comment (yonadaaa) [reply packages/world/src/WorldProxy.sol:46]: Sure, I'm neither here nor there. Cool to try them both

File: packages/world/src/WorldProxy.sol [hunk 3]
```diff
@@ -44,6 +40,8 @@ contract WorldProxy is Proxy {
     StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
 
     emit IERC1967.Upgraded(newImplementation);
```
Comment (holic) [inline packages/world/src/WorldProxy.sol:42]: is it okay that this is emitted even on a new deploy?

**Review by holic — APPROVED** [review]

File: packages/world/src/WorldProxy.sol (hunk 3 above)
Comment (yonadaaa) [reply packages/world/src/WorldProxy.sol:42]: Yes that's part of the spec here https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Proxy.sol#L26-L28

**Pull request merged**